app.js

```diff
@@ -1,2 +1,126 @@
 /* add your code here */
+document.addEventListener('DOMContentLoaded', () => {
+    const stocksData = JSON.parse(stockContent);
+    const userData = JSON.parse(userContent);
 
+    // Get the save and delete buttons
+
+    const deleteButton = document.querySelector('#btnDelete');
+	const saveButton = document.querySelector('#btnSave');
+
+    generateUserList(userData, stocksData); 
+    
+    // Register the event listener on the delete button
+
+    deleteButton.addEventListener('click', (event) => {
+	     event.preventDefault();
+         
+   // Find the index of the user in the data array
+
+		 const userId = document.querySelector('#userID').value;
+		 const userIndex = userData.findIndex(user => user.id == userId);
+         
+         // Remove the user from the array
+		 userData.splice(userIndex, 1);
+         // Render the updated user list
+		 generateUserList(userData, stocksData);
+	})
+
+    // Register the event listener on the save button
+	saveButton.addEventListener('click', (event) => {
+	     event.preventDefault();
+		 const id = document.querySelector('#userID').value;
+		 const userIndex = userData.findIndex(user => user.id == id);
+
+         const newUsers = [
+            ...userData.slice(0, userIndex),
+            {
+                 user: {
+                      firstname: document.querySelector('#firstname').value,
+                      lastname: document.querySelector('#lastname').value,
+                      address: document.querySelector('#address').value,
+                      city: document.querySelector('#city').value,
+                      email: document.querySelector('#email').value,
+                 },
+            },
+            ...userData.slice(userIndex + 1)
+        ];
+
```

Comment:
We need to make sure we are assigning `newUsers` to the `userData` variable here:
```js
userData = newUsers;
```